app/src/main/java/org/ea/sqrl/utils/PasswordStrengthMeter.java

```diff
@@ -0,0 +1,230 @@
+                ImageView imgLowercase = mPassStrengthLayout.findViewById(R.id.imgPasswordContainsLowercase);
+                ImageView imgDigits = mPassStrengthLayout.findViewById(R.id.imgPasswordContainsDigits);
+                ImageView imgSymbols = mPassStrengthLayout.findViewById(R.id.imgSymbols);
+                ProgressBar progressPwStrength = mPassStrengthLayout.findViewById(R.id.progressPasswordStrength);
+
+                progressPwStrength.setMax(STRENGTH_POINTS_MIN_GOOD);
+                progressPwStrength.setProgress(result.StrengthPoints);
+
+                switch (result.Rating) {
+
+                    case Poor:
+                        txtPasswordStrength.setText(mContext.getText(R.string.password_strength_poor));
+                        txtPasswordStrength.setBackgroundColor(
+                                ContextCompat.getColor(mContext, R.color.password_strength_poor));
+                        break;
+
+                    case Medium:
+                        txtPasswordStrength.setText(mContext.getText(R.string.password_strength_medium));
+                        txtPasswordStrength.setBackgroundColor(
+                                ContextCompat.getColor(mContext, R.color.password_strength_medium));
+                        break;
+
+                    case Good:
+                        txtPasswordStrength.setText(mContext.getText(R.string.password_strength_good));
+                        txtPasswordStrength.setBackgroundColor(
+                                ContextCompat.getColor(mContext, R.color.password_strength_good));
+                        break;
+                }
+
+                ImageView iv = null;
+                for (CharacterClass cc : result.CharClassCounts.keySet()) {
+                    if (cc == CharacterClass.Lowercase) iv = imgLowercase;
+                    if (cc == CharacterClass.Uppercase) iv = imgUppercase;
+                    if (cc == CharacterClass.Digit) iv = imgDigits;
+                    if (cc == CharacterClass.Symbol) iv = imgSymbols;
+
+                    int count = result.CharClassCounts.get(cc);
+
+                    Drawable drawable = count > 0 ?
+                            ContextCompat.getDrawable(mContext, R.drawable.led_green) :
+                            ContextCompat.getDrawable(mContext, R.drawable.led_red);
+
+                    if (iv != null) iv.setImageDrawable(drawable);
+                }
+
+                if (result.PasswordLength > 0 && result.PasswordLength < PW_MIN_LENGTH) {
+                    txtPasswordWarning.setText(R.string.short_password_warning);
+                    txtPasswordWarning.setVisibility(View.VISIBLE);
+                }
+                else txtPasswordWarning.setVisibility(View.GONE);
+            }
+            catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+
+    class PasswordStrengthResult {
+
+        public int StrengthPoints = 0;
```

Comment:
In Java, we usually start variables, members, fields, and functions with lowercase.

I usually always use the get/set mentality instead of public fields just so you can override these later with functionality. But in this case, as it's an internal class I think it's ok to skip this.

Reply:
Thanks for the input Daniel, very much appreciated!

I even considered doing it the "proper" way (get/set) but it just seemed so overly verbose for something that mundane.

As for Java coding conventions, my apologies for the occasional slip in that area, it's hard if you're mainly working in a different language and muscle memory kicks in.